scripts/conversion_script.py

```diff
@@ -598,17 +642,42 @@ def main():
 
         stdout, returncode = run_convert2rhel()
         conversion_successful = returncode == 0
-
+        rollback_errors = check_for_inhibitors_in_rollback()
+
+        # Returncode other than 0 can happen in two states in analysis mode:
+        #  1. In case there is another instance of convert2rhel running
+        #  2. In case of KeyboardInterrupt, SystemExit (misplaced by mistaked),
+        #     Exception not catched before.
+        # In any case, we should treat this as separate and give it higher
+        # priority. In case the returncode was non zero, we don't care about
+        # the rest and we should jump to the exception handling immediatly
         if not conversion_successful:
-            output.message = (
-                "An error occurred during the conversion execution. For details, refer to "
-                "the convert2rhel log file on the host at /var/log/convert2rhel/convert2rhel.log"
+            raise ProcessError(
+                message=(
+                    "An error occurred during the pre-conversion execution. For details, refer to "
+                    "the convert2rhel log file on the host at /var/log/convert2rhel/convert2rhel.log"
+                ),
+                report=(
+                    "convert2rhel execution exited with code %s"
+                    "Output of failed command: %s" % (returncode, stdout.rstrip("\n"))
+                ),
             )
-            output.report = (
-                "convert2rhel execution exited with code %s and output: %s."
-                % (returncode, stdout.rstrip("\n"))
+
+        # Check if there are any inhibitors in the rollback logging. This is
+        # necessary in the case where the analysis was done successfully, but
+        # there was an error in the rollback log.
+        if rollback_errors:
+            raise ProcessError(
+                message=(
+                    "A rollback of changes performed by convert2rhel failed. The system is in an undefined state. "
+                    "Recover the system from a backup or contact Red Hat support."
+                ),
+                report=(
+                    "\nFor details, refer to the convert2rhel log file on the host at "
+                    "/var/log/convert2rhel/convert2rhel.log. Relevant lines from log file: \n%s\n"
+                )
+                % rollback_errors,
```

Comment:
Can this happen in conversion script?

1. I run conversion which run pre-conversion check first
2. lets assume it ends in pre-conversion
3. then the conversion is unsuccessfull and rollbacks are not checked because the condition for exit status will be hit first

If I assume it won't end with pre-conversion and we will actually try for conversion - is there also rollback? (asking because I don't have the neccessary c2r knowledge). 

I guess in short ...is there a case where the conversion command would exit with non-zero but there would be rollback errors? :thinking:

Reply:
You're correct, @andywaltlova. When running a conversion (not analysis) and the pre-conversion analysis finds an inhibitor, convert2rhel exits with 1. So the current script implementation does not even get to the `if rollback_errors:` condition and raises an exception earlier under the `if not conversion_successful:` condition.

Reply:
Good point. Will update the code to reflect this suggestion. 

In my mind, I was considering that any non-zero exit code from convert2rhel would be more impactful than the rollback log.